pennylane/devices/default_qutrit.py

```diff
@@ -0,0 +1,261 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+The default.qutrit device is PennyLane's standard qutrit-based device.
+
+It implements the :class:`~pennylane._device.Device` methods as well as some built-in
+:mod:`qutrit operations <pennylane.ops.qutrit>`, and provides simple pure state
+simulation of qutrit-based quantum computing.
+"""
+import functools
+import numpy as np
+
+import pennylane as qml  # pylint: disable=unused-import
+from pennylane import QutritDevice
+from pennylane.wires import WireError  # pylint: disable=unused-import
+from .._version import __version__
+
+# tolerance for numerical errors
+tolerance = 1e-10
+
+OMEGA = np.exp(2 * np.pi * 1j / 3)
+
+
+class DefaultQutrit(QutritDevice):
```

Comment:
Inheriting from `DefaultQubit` does make sense from the standpoint of avoiding duplicate code. With that being said, I do think that it would be a bandaid solution, with the ideal solution being a more general parent class for both `DefaultQubit` and `DefaultQutrit`.

Reply:
I recall the consideration of creating such a general parent class, what was the reason for abandoning it?

Its need becomes relevant with qutrit/qudit simulations being added. The class could even be a mixin class that defines computing statistics given the dimension of the space used.

Reply:
I found the comments related to this and relevant parts of the ADR. It would be great to mention in the ADR that some logic of `QutritDevice` is identical to the logic in `QubitDevice` and `QutritDevice` and that we'd like to move those to a `QuditDevice` class down the line. Mentioning this explicitly will help us pick up this project in the future if need be.

The ADR document can simply be amended with a PR.

Reply:
@antalszava I will take of updating the ADR since Mudit does not have access to the repo.

Reply:
> I recall the consideration of creating such a general parent class, what was the reason for abandoning it?
The idea hasn't been abandoned. Qutrits were meant to be a specific example of higher dimensional simulation being possible on PennyLane, with the end goal being generalization into a `QuditDevice` and `DefaultQudit` device.

> It would be great to mention in the ADR that some logic of QutritDevice is identical to the logic in QubitDevice and QutritDevice and that we'd like to move those to a QuditDevice class down the line. Mentioning this explicitly will help us pick up this project in the future if need be.
This is a great point. I'll talk to @glassnotes and we will amend the ADR accordingly :)